modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ddl/DdlCommandHandler.java

```diff
@@ -352,59 +343,42 @@ private CompletableFuture<Boolean> handleCreateIndex(CreateIndexCommand cmd) {
                             "Can't create index on duplicate columns: " + String.join(", ", cmd.columns()));
                 });
 
-        Consumer<TableIndexChange> indexChanger = tableIndexChange -> {
-            switch (cmd.type()) {
-                case SORTED:
-                    createSortedIndexInternal(cmd, tableIndexChange.convert(SortedIndexChange.class));
+        boolean failIfExists = !cmd.ifNotExists();
 
-                    break;
-                case HASH:
-                    createHashIndexInternal(cmd, tableIndexChange.convert(HashIndexChange.class));
+        return catalogCreateIndexAsync(cmd)
+                .handle((unused, throwable) -> {
+                    if (throwable != null) {
+                        throwable = ExceptionUtils.unwrapCause(throwable);
```

Comment:
```suggestion
                .handle(handleModificationResult(!failIfExists, IndexAlreadyExistsException.class));
```

Reply:
We already have postprocessing method `handleModificationResult`.
Please, reuse it.

Reply:
fix it